driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -168,16 +174,20 @@ public Connection borrowConnection(long timeout, TimeUnit unit) throws Connectio
             throw new ConnectionException(host.getSocketAddress(), "Pool is " + phase);
 
         if (connections.isEmpty()) {
-            for (int i = 0; i < options().getCoreConnectionsPerHost(hostDistance); i++) {
-                // We don't respect MAX_SIMULTANEOUS_CREATION here because it's  only to
-                // protect against creating connection in excess of core too quickly
-                scheduledForCreation.incrementAndGet();
-                manager.blockingExecutor().submit(newConnectionTask);
+            if (!host.convictionPolicy.canReconnectNow())
+                throw new TimeoutException("Connection pool is empty, currently trying to reestablish connections");
+            else {
+                for (int i = 0; i < options().getCoreConnectionsPerHost(hostDistance); i++) {
```

Comment:
Answering my own question: **No** that can't happen since `newConnectionTask` ultimately calls `addConnectionIfUnderMaximum` which bails out if we are at or above maximum connections per hot.